src/MahApps.Metro/ThemeManager/ThemeManager.cs

```diff
@@ -374,10 +374,21 @@ public static bool IsThemeDictionary([NotNull] ResourceDictionary resources)
         /// <returns>The resource object or null, if the resource wasn't found.</returns>
         public static object GetResourceFromAppStyle(Window window, string key)
         {
-            var appStyle = (window.IsNull()
+            Theme appStyle;
+
+            if (Application.Current.IsNull()) // In case the Window is hosted in a WinForm app
```

Comment:
I have used the `IsNull()` extension method here as it seems to be used in this class. If needed I could extract the methods `IsNull()` and `IsNotNull()` into a separate class and use them for the other changes.

Reply:
I don't know what is better in these cases. Should we always use these 2 extension methods? Or not? I don't know...

Reply:
TBH I am not really sure to understand what those 2 methods bring compare to having the null check. Was it to avoid some FPs from ReSharper with the null reference rules?

Reply:
Yes this could be the reason, @batzen can explain this a little bit more.

Reply:
I added those because writing `if (!(xy is null))` feels horrible, whereas `if (xy.IsNotNull())` does not. And to be consistent i also added `IsNull`.

Reply:
@batzen I don't know whether this is a good practice or not but you could use `if (x is object)` to test non-nullability.

Reply:
just to complete this: Since C# 7.0 you could also use this code:
```
        if (SomeObject is SomeType variableName)
        {
            // Do some stuff with variableName
            return variableName.ToString();
        }
```

More about this: [https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/is](https://docs.microsoft.com/dotnet/csharp/language-reference/keywords/is)

Happy coding 

Tim

Reply:
@timunie That's correct, and I use it more and more, because of better readable code, but sometimes you don't need extra vars ;-)